src/components/Api.js

```diff
@@ -0,0 +1,95 @@
+export default class Api {
+    constructor({ baseUrl, headers }) {
+      this._baseUrl = baseUrl;
+      this._headers = headers;
+    }
+  
+    _handleResponse(res) {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    }
+  
+    _handleError(err) {
+      console.error(err);
+    }
+  
+    // User routes
+    getUserInfo() {
+      return fetch(`${this._baseUrl}/users/me`, {
+        headers: this._headers,
+      })
+        .then(this._handleResponse)
+        .catch(this._handleError);
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js` after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.